sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -325,7 +359,58 @@ private CassandraIO() {}
       checkArgument(entity() != null, "withEntity() is required");
       checkArgument(coder() != null, "withCoder() is required");
 
-      return input.apply(org.apache.beam.sdk.io.Read.from(new CassandraSource<>(this, null)));
+      try (Cluster cluster =
+          getCluster(hosts(), port(), username(), password(), localDc(), consistencyLevel())) {
+        if (isMurmur3Partitioner(cluster)) {
+          LOG.info("Murmur3Partitioner detected, splitting");
+
+          List<BigInteger> tokens =
```

Comment:
For maintenance coherence in DoFn based IOs we follow a pattern where the first part Creates the required object to query (the one I mentioned above), the second splits the query into multiples queries for each partition via a `SplitFn extends DoFn` and then with the result of such splits we do a read equivalent to current `QueryFn`. Can you refactor this segment to behave like that (with the query object I mentioned above).